storage/webdav/storage_webdav.go

```diff
@@ -0,0 +1,127 @@
+package webdav
+
+/*
+ * knoxite
+ *     Copyright (c) 2020, Fabian Siegel <fabians1999@gmail.com>
+ *
+ *   For license see LICENSE
+ */
+
+import (
+	"errors"
+	"net/url"
+
+	"github.com/studio-b12/gowebdav"
+
+	knoxite "github.com/knoxite/knoxite/lib"
+)
+
+//StorageWebDav stores data on a WebDav Server
+type StorageWebDav struct {
+	URL    url.URL
+	Client *gowebdav.Client
+	knoxite.StorageFilesystem
+}
+
+// Error declaration
+var (
+	ErrInvalidAuthentication = errors.New("Wrong Username or Password")
```

Comment:
We should probably move this to a central position as I can see more backends requiring it. Just as an example the FTP backend is also declaring one. Not a blocker for this PR, though.